docs/challenges.rst

```diff
@@ -0,0 +1,156 @@
+  with a header or footer, or serves them with an unusual MIME type.
+
+* (With standalone plugin)
+  You tried to use `--standalone` when there was already some other
+  program on your server listening to port 80
+
+* (With webroot plugin)
+  You tried to use `--webroot` when you don't have an existing web
+  server listening on port 80
+
+* Your DNS records aren't valid
+  Try checking your DNS records with a tool like the DNSchecker at
+  http://www.dnsstuff.com/ to make sure there are no serious errors.
+  Sometimes a DNS error still allows your site to load in a web
+  browser, but prevents the certificate authority from issuing a
+  certificate.
+
+TLS-SNI-01 challenge:
+* You aren't running Certbot on your web server
+
+  Most people should install and run Certbot on their web server hosting their website, not on their laptops or some other computer.  While you can use Certbot in manual mode on a laptop and then separately set up the appropriate files on your webserver, it's not likely to be the most convenient way to get a certificate for most users.
+
+* A domain name you're requesting a certificate for isn't correctly
+  pointed at that web server
+
+  In most cases, every name you're requesting a certificate for should
+  already exist and be pointed to the server where you're requesting
+  that certificate.  (Some alternatives exist for complex network
+  configurations, but they're the exception rather than the rule.)
+
+* You're using a content delivery network (CDN)
+
+  TLS-SNI-01 doesn't work with CDNs (like CloudFlare and Akamai).  You
+  have to use a different challenge type.  (This is a special case of
+  the previous problem: the domain name is pointed at the CDN, not
+  directly at your server.)
+
+* A firewall is blocking access to port 443
+
+  The certificate authority needs to be able to connect to port 443 of
+  your server in order to confirm that you satisfied the TLS-SNI-01
+  challenge.  So that needs to be publicly reachable from the Internet,
+  and not blocked by a router or firewall.
+
+* (With apache plugin)
+  Certbot thinks you're running Apache, but you aren't running it, or
+  you're running a different server of some kind on port 443
+
+* (With nginx plugin)
+  Certbot thinks you're running nginx, but you aren't running it, or
+  you're running a different server of some kind on port 443
+
+* (With apache or nginx plugin)
+  Certbot doesn't know how to modify your web server configuration correctly
+
+* (With standalone plugin)
+  You tried to use `--standalone` when there was already some other
+  program on your server listening to port 443
+
+* Your DNS records aren't valid
+  Try checking your DNS records with a tool like the DNSchecker at
+  http://www.dnsstuff.com/ to make sure there are no serious errors.
+  Sometimes a DNS error still allows your site to load in a web
+  browser, but prevents the certificate authority from issuing a
+  certificate.
+
+DNS-01 challenge:
```

Comment:
Another possible issue with DNS-01 would be not waiting long enough for the DNS change to propagate.